core/src/main/java/org/elasticsearch/search/aggregations/bucket/filters/FiltersAggregator.java

```diff
@@ -118,24 +135,26 @@ public InternalAggregation buildEmptyAggregation() {
     }
 
     final long bucketOrd(long owningBucketOrdinal, int filterOrd) {
-        return owningBucketOrdinal * filters.length + filterOrd;
+        return owningBucketOrdinal * (filters.length + 1) + filterOrd;
```

Comment:
Maybe we can make it slightly more efficient by storing eg. a "totalNumKeys" parameter on the aggregator which is equal to filters.length when showOtherBucket is false and filters.length+1 otherwise. Otherwise if you do not need the other bucket (default), you would waste 1/filters.length of the bucket ord space